src/hydro_system.hpp

```diff
@@ -583,6 +571,112 @@ void HydroSystem<problem_t>::FlattenShocks(amrex::MultiFab const &q_mf, amrex::M
 	});
 }
 
+// to ensure that physical quantities are within reasonable
+// floors and ceilings which can be set in the param file
+template <typename problem_t>
+void HydroSystem<problem_t>::EnforceLimits(amrex::Real const densityFloor, amrex::Real const pressureFloor, amrex::Real const speedCeiling,
+					   amrex::Real const tempCeiling, amrex::Real const tempFloor, amrex::MultiFab &state_mf)
+{
+
+	amrex::Real const rho_floor = densityFloor; // workaround nvcc bug
+	amrex::Real const P_floor = pressureFloor;
+	auto state = state_mf.arrays();
+
+	amrex::ParallelFor(state_mf, [=] AMREX_GPU_DEVICE(int bx, int i, int j, int k) noexcept {
+		int dindex = HydroSystem<problem_t>::density_index;
```

Comment:
Since this function now resides within `HydroSystem`, you can just use `density_index`, `x1Momentum_index`, etc. directly.